demosplan/DemosPlanCoreBundle/Logic/ApiRequest/ResourceType/DplanResourceTypeTrait.php

```diff
@@ -0,0 +1,197 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * This file is part of the package demosplan.
+ *
+ * (c) 2010-present DEMOS plan GmbH, for more information see the license file.
+ *
+ * All rights reserved
+ */
+
+namespace demosplan\DemosPlanCoreBundle\Logic\ApiRequest\ResourceType;
+
+use DemosEurope\DemosplanAddon\Contracts\Config\GlobalConfigInterface;
+use DemosEurope\DemosplanAddon\Contracts\CurrentUserInterface;
+use demosplan\DemosPlanCoreBundle\EventDispatcher\TraceableEventDispatcher;
+use demosplan\DemosPlanCoreBundle\Logic\ApiRequest\JsonApiResourceTypeService;
+use demosplan\DemosPlanCoreBundle\Logic\ApiRequest\PrefilledResourceTypeProvider;
+use demosplan\DemosPlanCoreBundle\Logic\EntityWrapperFactory;
+use demosplan\DemosPlanCoreBundle\Logic\Procedure\CurrentProcedureService;
+use demosplan\DemosPlanCoreBundle\Logic\ResourceTypeService;
+use demosplan\DemosPlanCoreBundle\Logic\User\CustomerService;
+use EDT\DqlQuerying\ConditionFactories\DqlConditionFactory;
+use EDT\DqlQuerying\SortMethodFactories\SortMethodFactory;
+use EDT\JsonApi\RequestHandling\MessageFormatter;
+use EDT\Querying\Contracts\SortMethodFactoryInterface;
+use EDT\Wrapping\Contracts\TypeProviderInterface;
+use EDT\Wrapping\WrapperFactories\WrapperObjectFactory;
+use Psr\Log\LoggerInterface;
+use Symfony\Contracts\EventDispatcher\EventDispatcherInterface;
+use Symfony\Contracts\Service\Attribute\Required;
+use Symfony\Contracts\Translation\TranslatorInterface;
+
+/**
+ * Provides helper class properties and corresponding injection/initialization for {@link DplanResourceType}.
+ *
+ * Do not add any unrelated properties or methods to this trait.
+ */
+trait DplanResourceTypeTrait
```

Comment:
Not so sure whether such a huge trait is really useful. I think I would rather inline it. Where would you reuse this?

Reply:
The code in the trait is neither reusable as trait nor when inlined. The goal is to keep the `DplanResourceType` more maintainable and also discoverable by someone having to adjust it, than it would be if the code was inlined like it was before.